### PR TITLE
DM-16234: Move transform and object catalog tasks to pipe_tasks

### PR DESCRIPTION
**Pull request by yalsayyad**
Will squash obviously after review

### CODE REVIEW

**Comment by morriscb** [comment]
Oh, also I remember seeing `lsst.qa.explorer` in a few places around the code. Need to change those.

File: python/lsst/pipe/tasks/parquetTable.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+                return self._df[columns]
+
+        if columns is None:
+            return self._pf.read().to_pandas()
+
+        try:
+            df = self._pf.read(columns=columns, use_pandas_metadata=True).to_pandas()
+        except AttributeError:
+            columns = self._sanitizeColumns(columns)
+            df = self._pf.read(columns=columns, use_pandas_metadata=True).to_pandas()
```
[the remark this answers is not in the document]
Comment (yalsayyad) [reply python/lsst/pipe/tasks/parquetTable.py]: When testing, I found that in the latest version of pyarrow `pyarrow.parquet.ParquetFile` no longer throws an `AttributeError` if the requested column name is missing.  I'm going to remove the `except AttributeError` and just check that column lengths match by hand and warn on which columns aren't in the result if they don't.

[the remark this answers is not in the document]
Comment (yalsayyad) [reply python/lsst/pipe/tasks/parquetTable.py]: Changed my mind on changing behavior on this ticket. This behavior was intentionally specified in one of @timothydmorton's unit tests. I added some more unit tests to complete coverage of Parquet and MultilevelParquet for the cases where no good columns are requested and only some good columns are requested. Deferring behavior change to DM-21976: Decide on behavior of ParquetTable if one requested column does not exist.

[the remark this answers is not in the document]
Comment (morriscb) [reply python/lsst/pipe/tasks/parquetTable.py]: Cool, as long as there's a ticket to decide I'm happy.

File: python/lsst/pipe/tasks/functors.py [hunk 2]
```diff
@@ -0,0 +1,995 @@
...
+    The `_columnLevels` and `_dfLevels` attributes should generally not need to
+    be changed, unless `_func` needs columns from multiple filters or datasets
+    to do the calculation.
+    An example of this is the ``lsst.qa.explorer.functors.Color` functor, for which
+    `_dfLevels = ('filter', 'column')`, and `_func` expects the dataframe it gets to
+    have those levels in the column index.
+
+    While not currently implemented, it would be
+    relatively straightforward to generalize the base `Functor` class to be able to
+    accept arbitrary `ParquetTable` formats (other than that of `deepCoadd_obj`).
```
[the remark this answers is not in the document]
Comment (yalsayyad) [reply python/lsst/pipe/tasks/functors.py]: Looks like you tried it in  https://jira.lsstcorp.org/browse/DM-17999 and it wasn't terrible

**Pull request merged**